### PR TITLE
[WIP] feat(set-auth-token): support scoped registries

### LINKED ISSUE #26: Support Scoped Registries
`registry-url.js` only retrieves the value of `registry`, which means that it will always append the `_authToken` to the value of `registry`. Therefore, scoped registries in the `.npmrc` configuration file, such as `@company:registry`, are not supported.

In my use case I have an `.npmrc` configuration file with two registries in it:

```
registry = "internal mirror of public registry"
@company:registry = "private registry"
```

The project's `package.json` file has set `publishConfig` to publish to the scoped, `@company`, registry. The project's name is `@company/something`.

In my case, I believe the value of the environment variable `NPM_TOKEN` needs to be appended to `@company:registry` instead of `registry` to allow proper authentication when publishing the project.

If you have access to the project's name, you can call something like [registry-url](https://www.npmjs.com/package/registry-url) to get the appropriate scoped registry.

An example of how `registry-url` is used can be seen in [package-json](https://github.com/sindresorhus/package-json/blob/master/index.js#L10).

### PR DESCRIPTION
**Pull request by hbetts**
Closes #26

### CODE REVIEW

**Comment by hbetts** [comment]
This is a work in progress. Missing unit tests.

File: src/test/registry-url.js [hunk 1]
```diff
@@ -1,15 +0,0 @@
-gt.module('npm registry');
-var check = require('check-more-types');
-
-var registryUrl = require('../registry-url');
-
-gt.asyncTest('getting npm registry url', 2, function () {
```
Comment (bahmutov) [inline src/test/registry-url.js]: why did you remove this test? if we switch to 3rd party module, we still want to make sure it returns an url,
I think we should leave this test, just switch it to use to 'registry-url'

Comment (destroyerofbuilds) [inline src/test/registry-url.js]: `registry-url` already has an [extensive set of unit tests](https://github.com/sindresorhus/registry-url/blob/master/test.js) that verify the module's public API matches the expectations we have from the module's documentation.

So if the purpose of adding a unit test is to ensure `registry-url` returns the value of `registry` from an `.npmrc` file, the test would be redundant, thereby adding no value.

If the purpose is to verify `registry-url` returns a string that is a valid URL, well, `registry-url` does not actually guarantee that. It returns the value assigned to `registry` in `.npmrc`. That string can be anything.

I set `registry` in my `./npmrc` file to be an invalid URL and then used `registry-url` to get that value:

```node
> url = require('registry-url')();
'this:is:not:a:url/'
```

A better test would be to write an integration test that does the following:
* Creates tmp directory and changes working directory to that tmp directory.
* Generates `.npmrc` file with `registry` preset to a known valid URL.
* Calls `set-auth-token` to generate updated `.npmrc` file with `_authToken` parameter appended.
* Verify modified URL matches expectations.

The integration test is still not perfect since the auth URL format is a moving target with each npm release.

Comment (bahmutov) [inline src/test/registry-url.js]: you are correct, yet the purpose of this test is to verify that the expected interface to "registry-url" is still valid. We don't have to test exhaustively but at least the single test that was in place verified that it was a function, and returned a url was good. Not a deal breaker for me, of course

**Comment by hbetts** [comment]
@bahmutov I've added back a test for `registry-url`. I further enhanced the tests by creating an isolated test environment in which to create and extract a registry URL. (One of the tests keeps mangling my personal `.npmrc` file. Therefore I'll probably continue to extend the tests with isolated environments)

In the process of writing a test for `set-auth-token.js` I ran into an issue with `form-auth-token.js`.

Basically it does not have support for generating a registry URL with a scope. I've thought of either extending it's method signature to take a scope argument (likely a breaking change), or writing another, similar, method for use in `set-auth-token.js`.

However, before going down that path I would like to have your opinion on how to address the lack of scope support in `form-auth-token.js`.

**Comment by bahmutov** [comment]
I would change `form-auth-token` to support authed scoped - if this is a breaking change, so be it, but in the commit message just have a line `BREAKING CHANGE` so semantic release properly increments the major in the version.

**Comment by bahmutov** [comment]
will look into this today - we are trying to use your semantic-release-gitlab and need this for our own registry :)

**Comment by bahmutov** [comment]
merging the pull request to fix the tests
